sdk/eventhub/azure-eventhub/azure/eventhub/aio/_client_base_async.py

```diff
@@ -145,7 +145,7 @@ def __init__(
         fully_qualified_namespace: str,
         eventhub_name: str,
         credential: Union[
-            "AsyncTokenCredential", AzureSasCredential, AzureNamedKeyCredential
+            "AsyncTokenCredential", AzureSasCredential, AzureNamedKeyCredential, EventHubSharedKeyCredential
```

Comment:
I tried to do the same thing as I did in the sync one:
`CredentialType = Union[AsyncTokenCredential, AzureSasCredential, AzureNamedKeyCredential, EventHubSharedKeyCredential]`

however it's giving me weird pyllint error
```
INFO:root:Namespaces found for package azure-eventhub: ['azure.eventhub', 'azure.eventhub.amqp', 'azure.eventhub.aio', 'azure.eventhub._eventprocessor', 'azure.eventhub.extensions', 'azure.eventhub.aio._eventprocessor']
************* Module azure.eventhub.aio._client_base_async
azure/eventhub/aio/_client_base_async.py:141: [C0412(ungrouped-imports), ] Imports from package azure are not grouped
************* Module azure.eventhub.aio._producer_client_async
azure/eventhub/aio/_producer_client_async.py:91: [E0601(used-before-assignment), EventHubProducerClient.__init__] Using variable 'CredentialTypes' before assignment
************* Module azure.eventhub.aio._consumer_client_async
azure/eventhub/aio/_consumer_client_async.py:148: [E0601(used-before-assignment), EventHubConsumerClient.__init__] Using variable 'CredentialTypes' before assignment
```

also `generated api stub reports
```
    from ._client_base_async import EventHubSharedKeyCredential
  File "/mnt/vss/_work/1/s/sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/.tox/apistub/lib/python3.9/site-packages/azure/eventhub/aio/_client_base_async.py", line 150, in <module>
    class ClientBaseAsync(ClientBase):
  File "/mnt/vss/_work/1/s/sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/.tox/apistub/lib/python3.9/site-packages/azure/eventhub/aio/_client_base_async.py", line 155, in ClientBaseAsync
    credential: CredentialTypes,
NameError: name 'CredentialTypes' is not defined
Traceback (most recent call last):
  File "/mnt/vss/_work/1/s/sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/../../../eng/tox/run_apistubgen.py", line 71, in <module>
    check_call(cmds, cwd=args.work_dir)
  File "/opt/hostedtoolcache/Python/3.9.10/x64/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['apistubgen', '--pkg-path', '/mnt/vss/_work/1/s/sdk/eventhub/azure-eventhub-checkpointstoreblob-aio', '--out-path', '/mnt/vss/_work/1/a/azure-eventhub-checkpointstoreblob-aio']' returned non-zero exit status 1.
ERROR: InvocationError for command /mnt/vss/_work/1/s/sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/.tox/apistub/bin/python /mnt/vss/_work/1/s/sdk/eventhub/azure-eventhub-checkpointstoreblob-aio/../../../eng/tox/run_apistubgen.py -t /mnt/vss/_work/1/s/sdk/eventhub/azure-eventhub-checkpointstoreblob-aio -w . --out-path /mnt/vss/_work/1/a (exited with code 1)
________________
```

I will just go with this async fix first instead of spending too much time on fixing the mypy.

Reply:
it worked 👍 thanks!